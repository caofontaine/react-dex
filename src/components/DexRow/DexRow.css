td {
  text-align: center;
}

.sprite {
  height: 64px;
  width: 64px;
}
