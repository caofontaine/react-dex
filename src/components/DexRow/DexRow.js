import React from 'react';
import './DexRow.css';

const DexRow = ({ dexNum, name, type, region }) => {
	const types = type.split(',');
	if (types.length === 1) types.push(' ');
	return (
		<tr>
			<td>{dexNum}</td>

			<td><img class="sprite" src={`/img/normal/${region.toLowerCase()}/${name.toLowerCase()}.png`} alt={name.toLowerCase()} /></td>

			<td>{name}</td>
			{
				types.map(t => {
					return (<td>{t}</td>)
				})
			}

			<td><input type="checkbox" id={name.toLowerCase()} name={name.toLowerCase()} value={name}  /></td>
			<td><input type="checkbox" id={`shiny${name.toLowerCase()}`} name={`shiny${name.toLowerCase()}`} value={`Shiny${name}`} /></td>
		</tr>
	)
}

export default DexRow;

//<td><img class="sprite" src="img/normal/kanto/bulbasaur.png" alt="bulbasaur" /></td>
