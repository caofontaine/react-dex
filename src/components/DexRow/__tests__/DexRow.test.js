import React from 'react';
import { shallow } from 'enzyme';
import DexRow from '../DexRow';

it('Expect to render DexRow component', () => {
  expect(shallow(<DexRow />)).toMatchSnapshot();
});
