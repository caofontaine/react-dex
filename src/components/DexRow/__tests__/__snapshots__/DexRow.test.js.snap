// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render DexRow component 1`] = `ShallowWrapper {}`;
